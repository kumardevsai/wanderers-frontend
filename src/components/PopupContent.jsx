import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Figure, CardImg, Figcaption } from '../elements/figure';

import PopupAction from './PopupAction';

import { observer, inject } from 'mobx-react';

@inject('WanderersStore')
@observer
export default class PopupContent extends Component {
  static propTypes = {
    place: PropTypes.object.isRequired
  };

  render() {
    const place = this.props.place;
    const firstImage = place.place_images[0];

    return (
      <Figure>
        <CardImg
          src={firstImage ? firstImage.card_image : '/noimg.jpg'}
          alt={place.name}
        />
        <hr />
        <Figcaption>{place.name.toUpperCase()}</Figcaption>
        <PopupAction place={place} />
      </Figure>
    );
  }
}
