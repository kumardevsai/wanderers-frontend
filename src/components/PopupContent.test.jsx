import PopupContent from './PopupContent';

it('displays place popup without an image', () => {
  const place = {
    name: 'Fake place',
    place_images: []
  };

  const popup = render(<PopupContent place={place} />);
  expect(popup).toMatchSnapshot();
});
