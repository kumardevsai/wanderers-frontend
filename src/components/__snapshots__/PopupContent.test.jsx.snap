// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays place popup without an image 1`] = `
<figure
  class="sc-bdVaJa kBeBlx"
>
  <img
    alt="Fake place"
    class="sc-bwzfXH iOZTkB"
    src="/noimg.jpg"
  />
  <hr />
  <figcaption
    class="sc-htpNat kFzgIi"
  >
    FAKE PLACE
  </figcaption>
</figure>
`;
